### DISCUSSION TITLE
数组卡片里面采用被动模式，数组删除卡片，其它卡片受影响

### DISCUSSION THREAD

**Discussion by xuanxiao2013** [alibaba/formily#2987] [Q&A 2.x] [answered]
问题复现地址： https://codesandbox.io/s/gallant-fire-ls37cv?file=/App.tsx

步骤如下：

1. 添加条目第一个卡片，类型选择 A， 人群选择 a； 
2. 继续添加第二个卡片，类型选择 B， 人群选择 b； 
3. 删除第一个卡片， 为什么第二个卡片中的人群 值会清空；

问题：
删除第一个卡片，为什么会影响第二个卡片？

**Comment by janryWang** [comment, marked as answer]
你这是被动模式，怎么是主动模式呢？主动模式是从具体事件触发联动的，比如onFieldInputValueChange

**Reply by xuanxiao2013** [reply]
嗯，写错了，是被动模式。单个卡片里面依赖比较多的时候，被动模式很好的简化逻辑。但是删除单个卡片，好像是认为整个字段树的修改，其它的卡片被动模式逻辑也被触发了。结论是数组的情况(有删除)不建议使用被动模式？ 还是说使用被动模式，https://codesandbox.io/s/gallant-fire-ls37cv?file=/App.tsx 这个里面的问题可以怎么解决？

**Reply by janryWang** [reply]
你如果要针对某个字段是否输入去清空其他字段值，最好用onFieldInputValueChange，被动模式的话，就得监听field.inputValue

**Reply by HawkingLan** [reply]
> 你如果要针对某个字段是否输入去清空其他字段值，最好用onFieldInputValueChange，被动模式的话，就得监听field.inputValue

不讨论主动被动，现在的问题是，删除卡片1，会导致卡片2中的属性值被清空，期望结果是“删除卡片1，卡片2不受影响”，这个问题怎么解决？
